psi4/driver/driver_nbody_multilevel.py

```diff
@@ -0,0 +1,146 @@
+#
+# @BEGIN LICENSE
+#
+# Psi4: an open-source quantum chemistry software package
+#
+# Copyright (c) 2007-2022 The Psi4 Developers.
+#
+# The copyrights for code used from other parties are included in
+# the corresponding files.
+#
+# This file is part of Psi4.
+#
+# Psi4 is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, version 3.
+#
+# Psi4 is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License along
+# with Psi4; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# @END LICENSE
+#
+
+import numpy as np
+
+def prepare_results(self, client=None):
+    """
+    Use different levels of theory for different expansion levels
+    See kwargs description in driver_nbody.nbody_gufunc
+    :returns: *return type of func* |w--w| The data.
+    :returns: (*float*, :py:class:`~psi4.core.Wavefunction`) |w--w| data and wavefunction with energy/gradient/hessian set appropriately when **return_wfn** specified.
+    """
+    from psi4.driver.driver_nbody import _print_nbody_energy
+
+    ptype = self.driver
+    natoms = self.molecule.natom()
+    supersystem = {k: v for k, v in self.task_list.items() if k.startswith('supersystem')}
+
+    # Initialize with zeros
+    energy_result, gradient_result, hessian_result = 0, None, None
+    energy_body_contribution = {b: {} for b in self.bsse_type}
+    energy_body_dict = {b: {} for b in self.bsse_type}
+    if ptype in ['gradient', 'hessian']:
+        gradient_result = np.zeros((natoms, 3))
+    if ptype == 'hessian':
+        hessian_result = np.zeros((natoms * 3, natoms * 3))
+
+    # Get numerical label for supersystem tasks
```

Comment:
How do you determine an appropriate label?

Reply:
I added that it's an index label. I think this can be handled cleaner upon revision.